sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/EventProcessorClient/EventProcessorClientTests.cs

```diff
@@ -708,6 +708,70 @@ public async Task HandlerPropertiesCanBeSetAfterEventProcessorHasStopped()
             Assert.That(() => processor.ProcessErrorAsync += eventArgs => Task.CompletedTask, Throws.Nothing);
         }
 
+        /// <summary>
+        ///   Verifies functionality of the <see cref="EventProcessorClient.ProcessErrorAsync" />
+        ///   event.
+        /// </summary>
+        ///
+        [Test]
+        public async Task ProcessingStartsWithTheNextEventAfterTheCheckpoint()
+        {
+            var fqNamespace = "namespace";
+            var eventHub = "hub";
+            var consumerGroup = "consumerGroup";
+            var partitionId = "3";
+            var checkpointOffset = 5631;
+            var checkpoint = new Checkpoint(fqNamespace, eventHub, consumerGroup, partitionId, checkpointOffset, 0);
+            var mockStorage = new MockCheckPointStorage();
+            var mockConsumer = new Mock<EventHubConsumerClient>(consumerGroup, Mock.Of<EventHubConnection>(), default);
+            var mockProcessor = new Mock<EventProcessorClient>(mockStorage, consumerGroup, fqNamespace, eventHub, Mock.Of<Func<EventHubConnection>>(), default) { CallBase = true };
+            var completionSource = new TaskCompletionSource<bool>();
+
+            mockStorage
+                .Checkpoints.Add((fqNamespace, eventHub, consumerGroup, partitionId), checkpoint);
+
+            mockConsumer
+                .Setup(consumer => consumer.GetPartitionIdsAsync(It.IsAny<CancellationToken>()))
+                .ReturnsAsync(new[] { partitionId });
+
+            mockConsumer
+                .Setup(consumer => consumer.ReadEventsFromPartitionAsync(
+                    partitionId,
+                    EventPosition.FromOffset(checkpointOffset + 1),
```

Comment:
I'd prefer that this was less brittle, but I don't believe that it warrants moving the `EventPosition` type to a shared concept in order to gain access to its internal members (for reading the offset member value), nor splitting it into a partial class.

I don't see this logic being something subject to change, so I'm willing to accept the trade-off here.